### DISCUSSION TITLE
Glutinum + ReactComponent type provider?

### DISCUSSION THREAD

**Discussion by jkone27** [Zaid-Ajaj/Feliz#633] [Ideas]
would be awesome is somehow `Glutinum` could be chained with `ReactComponent` to act as a type provider for React defined types

```fsharp
type SomeLibComponent = GlutinumReactProvider<"default","some-npm-react-library-with-ts-definition">
//and this could then have all required props with types and so on..

Html.div [
       SomeLibComponent.Hero "Hello"
       SomeLibComponent.Jumbotron [ ... ]
]

**Comment by MangelMaxime** [comment]
Glutinum doesn't work as a TypeProvider but instead as a transpiler / code generator.

There are plugins to make Glutinum support Feliz style syntax and I should soon explore this direction. But because React component are using JavaScript API and also even more heavily relying on UtilityType Glutinum first need to works with normal JavaScript code.

By using code generator it also allows to work around several more complex problem, because an fully automatic transformation means being able to identify the entry point of a library, transpile several files at once, and so.

Plus we need TypeScript compiler for Glutinum, unless you want to re-implement TypeScript in .NET. There is probably a way to make a tool for calling TypeScript compiler from a TypeProvider and using serialisation to move from one world to another but this is far too early to explore that at the moment,
